pdata/pcommon/common.go

```diff
@@ -772,6 +791,18 @@ func (m Map) Upsert(k string, v Value) {
 	}
 }
 
+// UpsertEmpty inserts or updates an empty value to the map under given key
+// and return the updated/inserted value.
+func (m Map) UpsertEmpty(k string) Value {
```

Comment:
Let's do an investigation if we have usages for the map API. I kind of believe that it is a good time to maybe switch to `sync.Map` like API, call this `StoreX` and remove all the variants... what do you think? Probably separate effort, but want to investigate.

Reply:
I was also curious if we really need all of them so I did an initial investigation. Most of the usages are Upsert/Insert that can be easily replaced with Upsert/Store . There are just few cases where Insert or Update are really needed e.g. in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/attraction/attraction.go#L319. But they can be also be replaced with Get+Upsert/Store, it's just one more iteration which is fine probably

Reply:
also attraction will be removed and replace soon by the tql package/processor